src/main/java/org/dbos/apiary/procedures/postgres/shop/ShopGetItem.java

```diff
@@ -0,0 +1,21 @@
+package org.dbos.apiary.procedures.postgres.shop;
+
+import org.dbos.apiary.postgres.PostgresContext;
+import org.dbos.apiary.postgres.PostgresFunction;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ShopGetItem extends PostgresFunction {
+
+    public static int runFunction(PostgresContext ctxt, int personID, String searchText, int maxCost) throws Exception {
+        String[] items = ctxt.apiaryCallFunction("ShopESSearchItem", searchText, maxCost).getStringArray();
+        if (items.length > 0) {
+            int bestItem = Integer.parseInt(items[0]);
+            ctxt.apiaryCallFunction("ShopAddCart", personID, bestItem);
```

Comment:
It looks a bit weird to do a Postgres write inside a seemingly read-only function. I wonder if it is possible to separately investigate latency from Postgres vs. latency from Elasticsearch?